docs/overview.rst

```diff
@@ -238,55 +237,66 @@ You have two choices:
+    and can be installed with `pip install`.
 
   - it is possible to choose between the :term:`CPython ABI` and the
-    :term:`HPy Universal ABI` when compiling an extension module
+    :term:`HPy Universal ABI` when compiling an extension module.
 
   - extensions compiled with the CPython ABI work out of the box on
-    CPython
+    CPython.
 
   - it is possible to load HPy Universal extensions on CPython, thanks to the
-    ``hpy.universal`` package
+    ``hpy.universal`` package.
 
   - it is possible to load HPy Universal extensions on
-    PyPy (using the PyPy `hpy branch <https://foss.heptapod.net/pypy/pypy/tree/branch/hpy>`_)
+    PyPy (using the PyPy `hpy branch <https://foss.heptapod.net/pypy/pypy/tree/branch/hpy>`_).
 
   - it is possible to load HPy Universal extensions on `GraalPython
-    <https://github.com/graalvm/graalpython>`_
+    <https://github.com/graalvm/graalpython>`_.
 
 
 However, there is still a long road before HPy is usable for the general
 public. In particular, the following features are on our roadmap but have not
 been implemented yet:
 
-  - it is not possible to write custom types (like NumPy's ndarray)
-    in C. There is already a WIP branch to address this issue
+  - many of the original Python/C functions have not been ported to
+    HPy yet. Porting most of them is straighforward, so for now the priority
+    is to work on the "hard" features to prove that the HPy approach works, and
+    we will port new functions as needed
 
-  - only a handful of the original Python/C functions have been ported to
-    HPy. Porting most of them is straighforward, so for now the priority is to
-    work on the "hard" features to prove that the HPy approach works, and we
-    will port new functions as needed
+  - handles are intended to be short-lived, but sometimes one needs a
+    long-lived reference to a Python object. In HPy, we call this
+    long-lived reference an `HPyField`, but we still need to implement it.
+    We also need `HPy_Store` and `HPy_Load` to save and load these fields.
 
-  - the debug mode simply does not exist (yet!)
+  - add C-level module state. Often an extension needs module state that is
+    accessible from C (e.g. if a module implements a new type `ArrayType`,
+    many of the module methods written in C may need access to `ArrayType`)
+    but HPy does not have convenient support for this yet.
 
-  - there is no standard/easy way to integrate HPy into a
-    ``distutils/setuptools`` based workflow. The only way to compile HPy
-    extensions right now is to manually clone the git repo and tweak your
-    ``setup.py``. Eventually, we plan to offer a workflow which integrates
-    seamlessly with ``pip``, ``setuptools``, etc.
+  - improve the debug mode so that it returns a C traceback showing where
+    a handle was leaked or a closed handle was used.
 
   - there is no integration with Cython. The medium-term plan is to extend
```

Comment:
Other Python compilers could also be mentioned here. When HPy will be usable with Numpy, it will also be good to extend/modify [Pythran](https://github.com/serge-sans-paille/pythran/) to generate HPy code. Note that since Pythran C++ only uses the CPython C-API for converting Python object to pure C++ objects, it should be much simpler than extending Cython (yes, also much less useful given how much Cython is used everywhere!).

Reply:
I agree it might be an interesting goal. I am not sure about putting it on the roadmap though, because AFAIK nobody of the core team is going to work on that anytime soon